src/vaadin-message.js

```diff
@@ -151,17 +152,45 @@ class MessageElement extends ElementMixin(ThemableMixin(PolymerElement)) {
     `;
   }
 
-  ready() {
-    super.ready();
-  }
-
   static get is() {
     return 'vaadin-message';
   }
 
   static get version() {
     return '2.0.0-alpha1';
   }
+
+  ready() {
+    super.ready();
+
+    // Handle focus
+    this.addEventListener('focus', () => this._setFocused(true), true);
+    this.addEventListener('blur', () => this._setFocused(false), true);
+    this.addEventListener('mousedown', () => {
+      this._mousedown = true;
+      const mouseUpListener = () => {
+        this._mousedown = false;
+        document.removeEventListener('mouseup', mouseUpListener);
+      };
+      document.addEventListener('mouseup', mouseUpListener);
+    });
+  }
+
+  /**
+   * @param {boolean} focused
+   * @protected
+   */
+  _setFocused(focused) {
+    if (focused) {
+      this.setAttribute('focused', '');
+      if (!this._mousedown) {
+        this.setAttribute('focus-ring', '');
+      }
+    } else {
+      this.removeAttribute('focused');
+      this.removeAttribute('focus-ring');
```

Comment:
`focus-ring` is missing tests. Currently one test checks that focus-ring is not set when message is clicked. There should be also tests for:
- focus-ring is set when focusing with keyboard
- focus-ring is removed on blur

Reply:
- I split the mouse test in two, so we have one that explicitly test that it is not set on mouse interaction
- I added a test in keyboard navigation that checks that focus-ring is added
- I did not directly find a way to get focus outside of the message list once it has been set. Any ideas on this? I tried calling `.focus()` on messageList and document. firing mousedown->focus->mouseup on messageList and document. pressing tab key on the currently selected item. None of these cleared `focused` or `focus-ring`.

Reply:
Try dispatching `blur` event manually.

Reply:
That worked \o/

Reply:
Done.